mmdet/evaluation/metrics/openimages_metric.py

```diff
@@ -1,237 +1,131 @@
-        results should be stored in ``self.results``, which will be used to
-        compute the metrics when all batches have been processed.
+        """Process one batch of data samples and predictions.
+
+        Parse predictions and ground truths from ``data_samples`` and invoke
+        ``self.add``.
 
         Args:
             data_batch (dict): A batch of data from the dataloader.
             data_samples (Sequence[dict]): A batch of data samples that
                 contain annotations and predictions.
         """
+        predictions, groundtruths = [], []
         for data_sample in data_samples:
-            gt = copy.deepcopy(data_sample)
-            # add super-category instances
-            # TODO: Need to refactor to support LoadAnnotations
-            instances = gt['instances']
-            if self.get_supercategory:
-                supercat_instances = self._get_supercategory_ann(instances)
-                instances.extend(supercat_instances)
-            gt_labels = []
-            gt_bboxes = []
-            is_group_ofs = []
-            for ins in instances:
-                gt_labels.append(ins['bbox_label'])
-                gt_bboxes.append(ins['bbox'])
-                is_group_ofs.append(ins['is_group_of'])
-            ann = dict(
-                labels=np.array(gt_labels, dtype=np.int64),
-                bboxes=np.array(gt_bboxes, dtype=np.float32).reshape((-1, 4)),
-                gt_is_group_ofs=np.array(is_group_ofs, dtype=bool))
-
-            image_level_labels = gt.get('image_level_labels', None)
-            pred = data_sample['pred_instances']
-            pred_bboxes = pred['bboxes'].cpu().numpy()
-            pred_scores = pred['scores'].cpu().numpy()
-            pred_labels = pred['labels'].cpu().numpy()
-
-            pred_bboxes, pred_scores, pred_labels = self._process_predictions(
-                pred_bboxes, pred_scores, pred_labels, instances,
-                image_level_labels)
-
-            dets = []
-            for label in range(len(self.dataset_meta['CLASSES'])):
-                index = np.where(pred_labels == label)[0]
-                pred_bbox_scores = np.hstack(
-                    [pred_bboxes[index], pred_scores[index].reshape((-1, 1))])
-                dets.append(pred_bbox_scores)
-            self.results.append((ann, dets))
-
-    def compute_metrics(self, results: list) -> dict:
-        """Compute the metrics from processed results.
+            pred = {
+                'bboxes': data_sample['pred_instances']
+                ['bboxes'].cpu().numpy(),  # noqa: E501
+                'scores': data_sample['pred_instances']
+                ['scores'].cpu().numpy(),  # noqa: E501
+                'labels':
+                data_sample['pred_instances']['labels'].cpu().numpy()
+            }
+            predictions.append(pred)
+
+            gt = {
+                'instances': data_sample['instances'],
```

Comment:
Can we get gt from the data_sample['gt_instances'] by adding 'LoadAnnotation' pipeline?
So that the gt and pred can be the same format.

![image](https://user-images.githubusercontent.com/12907710/192729669-0b0cf474-b9de-4d72-a8e7-5045cbea6640.png)